src/hotspot/share/jvmci/jvmciCodeInstaller.cpp

```diff
@@ -945,6 +945,10 @@ JVMCI::CodeInstallResult CodeInstaller::initialize_buffer(CodeBuffer& buffer, bo
     }
   }
 #endif
+  if (_has_auto_box) {
+    JavaThread* THREAD = JavaThread::current();
+    JVMCI::ensure_box_caches_initialized(JVMCI_lock, CHECK_(JVMCI::ok));
```

Comment:
Why we can hold JVMCI_lock here.
Does it mean that we'll never initialize the box classes at this point?

Reply:
I'm not sure I understand your question - why would we not be able to hold JVMCI_lock here?
That said, based on @veresov 's [earlier comment](https://github.com/openjdk/jdk/pull/1727#issuecomment-742381965), I'm now inclined to remove the lock altogether. As he says, at worst we harmlessly execute the loop in `ensure_box_caches_initialized` a few extra times. I've pushed this change.

Reply:
I think we should not hold JVMCI_lock here, because:

```
  If JVMCI_lock was acquired, then no_safepoint_verifier(new_owner, true) was called [1].
     --> Then, thread->_no_safepoint_count was increased [2].
         --> Then, the assert [3] would fail due to _no_safepoint_count > 0.
```

What do you think?
Thanks.

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/mutex.cpp#L480
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/mutex.cpp#L446
[3] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/thread.cpp#L795

Reply:
Yes, you're right - a VM lock should not be held when leaving the VM to call Java code.
As stated above, I've removed the lock altogether now (91571c1b3783fe2a2591d65a62fbf8286be173d9).